ledger/src/blockstore.rs

```diff
@@ -1838,6 +1852,75 @@ impl Blockstore {
             .map(|x| x.0)
     }
 
+    fn verify_shred(&self, leader: &Pubkey, slot: Slot, x: u64) {
+        let payload = self.get_data_shred(slot, x).unwrap().unwrap();
+        let shred = Shred::new_from_serialized_shred(payload).unwrap();
+        assert!(
+            shred.verify(leader),
+            "shred failed verification {} {} {}",
+            slot,
+            x,
+            self.shreds_dir,
+        );
+    }
+
+    fn verify_expected_shreds(
+        &self,
+        ls: Option<&Arc<LeaderScheduleCache>>,
+        slot: Slot,
+        total: &mut usize,
+    ) -> ShredIndex {
+        let index = self
+            .index_cf
+            .get(slot)
+            .unwrap()
+            .unwrap_or_else(|| Index::new(slot));
+        let data = index.data();
+        if slot != 0 {
+            if let Some(ls) = ls {
+                let leader = ls.slot_leader_at(slot, None).unwrap();
+                for ix in data.index.iter() {
+                    self.verify_shred(&leader, slot, *ix);
+                    *total += 1;
+                }
+            }
+        }
+        data.clone()
+    }
+
+    pub fn reconcile_shreds(
```

Comment:
Could we add a simple test that shows `reconcile_shreds` panics when the `Index` contains a data file that isn't on disk?

A unit test that checks for panics could use (taken from https://stackoverflow.com/questions/26469715/how-do-i-write-a-rust-unit-test-that-ensures-that-a-panic-has-occurred):
```
let result = std::panic::catch_unwind(|| <expected_to_panic_operation_here>);
    assert!(result.is_err());  //probe further for specific error type here, if desired
```